src/libcollections/str.rs

```diff
@@ -1739,27 +1739,50 @@ impl str {
         UnicodeStr::grapheme_indices(&self[..], is_extended)
     }
 
-    /// An iterator over the non-empty words of `self`.
+    /// An iterator over non-whitespace substrings of `self`,
+    /// separated by any sequence of whitespace.
     ///
-    /// A 'word' is a subsequence separated by any sequence of whitespace.
-    /// Sequences of whitespace
-    /// are collapsed, so empty "words" are not included.
+    /// Sequences of whitespace are collapsed, so empty substrings
```

Comment:
This function iterates over substrings separated by any sequence of whitespace, therefore this sentence is redundant.